code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -532,9 +532,13 @@
 		remove_reagent(reagent.type, transfer_amount)
 		transfer_log[reagent.type] = list(REAGENT_TRANSFER_AMOUNT = transfer_amount, REAGENT_PURITY = reagent.purity)
 
+	//combat log
 	if(transferred_by && target_atom)
+		if(isorgan(target_atom))
+			var/obj/item/organ/organ_item = target_atom
+			target_atom = organ_item.owner
 		target_atom.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
-		log_combat(transferred_by, target_atom, "transferred reagents ([get_external_reagent_log_string(transfer_log)]) from [my_atom] to")
+		log_combat(transferred_by, target_atom, "transferred reagents to", my_atom, "which had [get_external_reagent_log_string(transfer_log)]")
```

Comment:
Make a var called like "log_target" or something here, we shouldn't modify the value of target_atom outside this if check